frontend/src/app/settings/page.tsx

```
@@ -241,7 +481,16 @@ export default function SettingsPage() {
   }
 
   return (
-    <div className="bg-gradient-to-br from-purple-50 via-white to-violet-50 dark:from-gray-900 dark:via-gray-800 dark:to-purple-900">
+    <div className="bg-gray-50 dark:bg-gray-900 min-h-screen relative">
+      {/* Dark Mode Toggle - Fixed on right edge */}
+      <button
+        onClick={toggleTheme}
+        className="fixed right-4 top-1/2 -translate-y-1/2 z-50 p-3 rounded-full bg-white dark:bg-gray-800 shadow-lg hover:shadow-xl transition-all duration-300 border border-gray-200 dark:border-gray-700"
+        aria-label="Toggle dark mode"
+      >
```

Biome (2.1.2)

[error] Provide an explicit type prop for the button element.

The default type of a button is submit, which causes the submission of a form when placed inside a `form` element. This is likely not the behaviour that you want inside a React application.
Allowed button types are: submit, button or reset

(lint/a11y/useButtonType)